datamodel/include/data_types/math/Matrix.hpp

```diff
@@ -135,24 +136,64 @@ class Matrix
     // row-major
     const value_type &at(size_t row, size_t col) const
     {
-        return _m.at(row * DimRow + col);
+        return _m.at(row * rows + col);
     }
 
-    Matrix transpose() const
+    Matrix transposed() const
     {
         storage_type transposed_source {};
         for (size_t i = 0; i < rows; ++i)
         {
             const auto col = getColumn(i);
             for (size_t j = 0; j < col.dimension; ++j)
             {
-                transposed_source[i * DimRow + j] = col.at(j);
+                transposed_source[i * rows + j] = col.at(j);
             }
         }
 
         return Matrix(std::make_move_iterator(transposed_source.begin()), std::make_move_iterator(transposed_source.end()));
     }
 
+    template <typename T = void, typename = std::enable_if_t<rows == columns, T>>
+    [[nodiscard]] value_type determinant() const {
+        std::vector<value_type> param;
```

Comment:
Consider rewriting all the algorithms (determinant and inverse) using static storage (i.e. `std::array` instead of `std::vector`)

Reply:
This will result in a code-bloat even for lower order matrices. And if we increase a number of elemnts, it will be even worse. I think it well be better to leave a vector implementation.

Reply:
I will leave a TODO as a consideration for future optiomization if memory usage is too much.

Reply:
This will result in a code-bloat even for lower order matrices. And if we increase a number of elemnts, it will be even worse. I think it well be better to leave a vector implementation.

Reply:
I will leave a TODO as a consideration for future optiomization if memory usage is too much.

Reply:
Ok, fair enough.